### ISSUE TITLE
제목: [Day 4] CNN으로 CIFAR-10 분류 완료 - {김관태}

### ISSUE THREAD

**Issue by gonatae2-hash** [J00NQ/algorithm_YH#9]
## 실습 완료 내용

- [x] Kaggle Learn Lesson 1~3 완료 (The Convolutional Classifier, Convolution and ReLU, Maximum Pooling)

- [x] CIFAR-10 데이터 로드 & 탐색

- [x] CNN 모델 구축 (Conv2D + MaxPooling + Dense)

- [x] 모델 훈련 (조기 종료 적용)

- [x] 성능 평가 및 혼동행렬 분석

- [x] Dense Network와 성능 비교

## 데이터셋 정보

- 이미지 크기: 32×32 (컬러)

- 훈련 데이터: 40,000장

- 검증 데이터: 10,000장

- 테스트 데이터: 10,000장

- 클래스: 10가지 (airplane, automobile, bird, ..., truck)

## 모델 구축

- Conv2D(32, 3×3) + ReLU + MaxPooling(2×2)

- Conv2D(64, 3×3) + ReLU + MaxPooling(2×2)

- Conv2D(128, 3×3) + ReLU + MaxPooling(2×2)

- Flatten()

- Dense(256) + ReLU + Dropout(0.5)

- Dense(10) + Softmax (10-class 분류)

## 성능 결과

- 훈련 정확도: ____% 

- 검증 정확도: ____% 

- 테스트 정확도: ____% (최종)

- 총 파라미터: ____개

- 훈련 에포크: ____개 (조기 종료 적용)

## Dense Network와의 비교

| 항목 | Dense | CNN |

|------|-------|-----|

| 입력 형태 | 1D 벡터 | 2D 이미지 |

| 파라미터 수 | 약 77M | ____개 |

| 정확도 | 70~78% | ___% |

| 훈련 시간 | 2~5분 | ____분 |

| 메모리 | ~2GB | ____MB |

## 관찰 기록

1. CNN이 Dense Network보다 효율적인 이유:

   - 파라미터 감소: 

   - 공간 정보 보존:

   - 특징 학습의 계층성:

2. 가장 잘 분류되는 클래스 Top 3:

   - ______ (정확도: ___%)

   - ______ (정확도: ___%)

   - ______ (정확도: ___%)

3. 가장 어려운 클래스 (정확도 낮음):

   - ______ (정확도: ___%) → 이유:

   - ______ (정확도: ___%) → 이유:

4. 가장 많이 혼동되는 클래스 쌍:

   - ______ → ______ (____건)

   - ______ → ______ (____건)

## 개선 시도 (선택)

- 시도 1: (예: 필터 개수 증가) → 결과: ____%

- 시도 2: (예: Batch Normalization 추가) → 결과: ____%

- 가장 효과 있었던 변경: 

## 다음 단계 (내일)

Day 5: YOLO로 객체 검출 도전 예정

## 커밋 및 링크

- Kaggle 노트북: (URL)

- GitHub: (링크)

**Comment by gonatae2-hash** [comment]
<img width="131" height="76" alt="Image" src="https://github.com/user-attachments/assets/201fc820-def3-4400-9984-6fa7d05bb02e" />
<img width="541" height="181" alt="Image" src="https://github.com/user-attachments/assets/958dda13-69ca-496d-b84c-29b7a7f7b60a" />
<img width="351" height="279" alt="Image" src="https://github.com/user-attachments/assets/c1082d47-25eb-4a71-bd54-cc4a02bc800e" />
<img width="597" height="602" alt="Image" src="https://github.com/user-attachments/assets/74ef35b1-fe5e-41f1-b851-8684ca43fa80" />
